modules/SocketManager/module/src/socketmanager.c

```diff
@@ -773,6 +773,12 @@ process_sockets(ind_soc_priority_t priority)
         read_ready = (pfd->revents & POLLIN) != 0;
         write_ready = (pfd->revents & POLLOUT) != 0;
         error_seen = (pfd->revents & POLLERR) != 0;
+        /* POLLHUP only applied on socket. See poll(3) manpage.
+         * Let the callback to handle the error.
+         */
+        if (pfd->revents & POLLHUP) {
+            error_seen = POLLHUP;
+        }
```

Comment:
how about ```error_seen = pfd->revents & (POLLERR | POLLHUP);``` ?

also, error_seen used to be a boolean, so now it looks like all socket callbacks need to be reexamined to make sure they handle all possible values of error_seen properly.

Reply:
let's keep the comment, but replace lines 775-785 with ```error_seen = (pfd->revents & (POLLERR | POLLHUP | POLLNVAL));```